tests/test_cases/issue_348/issue_348.py

```diff
@@ -34,43 +30,40 @@ def __init__(self, edge, signal):
         self.monitor_edges = [0, 0]
         self.signal = signal
 
-    @cocotb.coroutine
-    def signal_mon(self, signal, idx, edge):
+    async def signal_mon(self, signal, idx, edge):
         while True:
-            yield edge(signal)
+            await edge(signal)
             self.monitor_edges[idx] += 1
 
-    @cocotb.coroutine
-    def start(self):
+    async def start(self):
         clock_edges = 10
 
         cocotb.fork(clock_gen(self.signal, clock_edges))
-        first = cocotb.fork(self.signal_mon(self.signal, 0, self.edge_type))
-        second = cocotb.fork(self.signal_mon(self.signal, 1, self.edge_type))
+        _ = cocotb.fork(self.signal_mon(self.signal, 0, self.edge_type))
+        _ = cocotb.fork(self.signal_mon(self.signal, 1, self.edge_type))
```

Comment:
Should we just drop the assignment here?

Reply:
```suggestion
        cocotb.fork(self.signal_mon(self.signal, 0, self.edge_type))
        cocotb.fork(self.signal_mon(self.signal, 1, self.edge_type))
```